### PR TITLE
[Arma 3] Egg Rewrite

### PR DESCRIPTION
**Pull request by redthirten**
# Description

> [!WARNING]
> This Egg will not function properly unless it's [sister pull request](https://github.com/pelican-eggs/yolks/pull/293) in the [Yolks repo](https://github.com/pelican-eggs/yolks) is also accepted and the image is compiled/published. For testing purposes, you may utilize this copy of the image: `ghcr.io/redthirten/arma3-yolk:main`

I have decided to (almost) completely rewrite my Arma 3 Egg and Yolk. This PR aims to:
- Solve numerous issues that have been brought to my attention over the years (namely on Discord, so I unfortunately can't link them easily)
- Remove hacky workarounds that are no longer needed.
- Clean up the Egg & Yolk in general and make it easier to use.

**I encourage people with real "production" Arma 3 servers to please help me beta test these changes. While I tried my best, Arma 3 is very complex and configurable, so it is hard to test for every edge case.**

### Egg Changelog:
- Made the default `server.cfg` download URL to be a static variable in the install script (instead of being local reference of the `BASIC_URL` variable). It can be modified by experienced hosts if they wish, but it did not warrant a whole Startup Variable.
- Updated the default "[Advanced] basic.cfg URL" value to point to the new repo and changed the variable name to `BASIC_CFG_URL` to be more clear. Note: This must remain a Startup Variable as it is referenced by the Yolk's entrypoint as well.
- Fixed a niche security concern by changing the user-editable "[Advanced] Extra Flags for SteamCMD" Startup Variable. Previously, a malicious end-user could input a specially crafted string to try and escape the SteamCMD process during a re-install and gain arbitrary code execution (ACE) on the install container. This variable, previously intended for installing the `profiling` branch, is now a more restricted "Arma 3 Branch (CDLC, etc.)" variable (which also replaces the "Download CDLC" variable).
- On a similar note, the "Extra Startup Parameters" Startup Variable has also been removed to prevent escaping the Arma 3 server process and gaining ACE on the server container. Most popular Arma startup parameters have been added as individual Startup Variables now.
- An alternate branch (eg. `creatordlc`) can now be installed on the first install instead of on the first Automatic Update.
- Removed the "[Advanced] Clear HC Profiles Cache on Startup" Startup Variable as it is no longer needed (ie. the issue it was fixing as been resolved).
- Fixed profiles path by allowing the server to just use its default path relative to the server executable (which is currently the only available method on Linux according to the docs).
- Fixed BattlEye path by allowing the server to just use its default path relative to the server executable.
- Added BattlEye RCon config creation to the Install Script, as well as Startup Variables and Configuration Files for easy editing.
- Updated the Startup Command to use a Yolk-generated parameter file. This should hopefully avoid any further issues with mod or parameter parsing in the future, as trying to organize and escape everything correctly is too much of a pain to maintain. This also future-proofs the Egg if Pelican decides to stop supporting Yolk-generated variables in the Startup Command (ie. pre-evaluated Startup Commands).
- Cleaned up Startup Variable names, descriptions, and ordering.
- Added update URL (for Pelican Egg).

### README Changelog:
- Updated Egg capabilities.
- Cleaned up Installation/System Requirements and added more pertinent information. Also moved it above the Server Ports section.
- Added note that the Egg no longer supports Wisp.
- Cleaned up and improved Server Ports section.
- Updated knowledge about mounting Workshop mods.
- General cleanup / verbiage improvements.

## Checklist for all submissions

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/pelican-eggs/games-steamcmd/blob/main/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
* [x] Have you tested and reviewed your changes with confidence that everything works?
* [x] Did you branch your changes and PR from that branch and not from your master branch?

* [x] You verify that the start command applied does not use a shell script
* [x] The egg was exported from the panel

### CODE REVIEW

**Comment by lajczi** [comment]
good job

File: arma/arma3/egg-arma3.json [hunk 1]
```diff
@@ -2,9 +2,9 @@
     "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PANEL",
     "meta": {
         "version": "PTDL_v2",
```
Comment (QuintenQVD0) [inline arma/arma3/egg-arma3.json:4]: This is not pelican exported or a realy old version

**Review by QuintenQVD0 — DISMISSED** [review]

File: arma/arma3/egg-arma3.json (hunk 1 above)
Comment (redthirten) [reply arma/arma3/egg-arma3.json:4]: I can't get file uploads (of any kind) to work on my current Pelican deployment in Docker. Would you mind possibly importing the Pterodactyl version on yours, exporting it as Pelican, and committing it to the branch? On a side note, unless this is already in the works, I might try to develop a Python script to convert Eggs both ways. This way, contributors that don't have both panels can easily make PRs that update both Eggs reliably.

**Comment by JasonTurk** [comment]
This lacks the ability to change the default ports?

**Comment by redthirten** [comment]
@JasonTurk No? Please read the README.

**Comment by redthirten** [comment]
@JasonTurk Why does there need to be an entry under Startup Variables if the Egg can just reference the Primary port assigned to the server instance?

**Review by gOOvER — APPROVED** [review]
Red said, it work
